lib/engine.js

```diff
@@ -397,10 +397,11 @@ module.exports = function (s, conf) {
             trade_balance = n(s.balance.currency).divide(100).multiply(buy_pct)
             tradeable_balance = n(s.balance.currency).divide(100 + fee).multiply(buy_pct)
             expected_fee = n(trade_balance).subtract(tradeable_balance).format('0.00000000', Math.ceil) // round up as the exchange will too
+            console.log(`s.product.asset_increment: ${s.product.asset_increment}`)
             if (buy_pct + fee < 100) {
-              size = n(tradeable_balance).divide(price).format('0.00000000')
+              size = n(tradeable_balance).divide(price).format(s.product.asset_increment ? s.product.asset_increment : '0.00000000', Math.floor)
             } else {
-              size = n(trade_balance).subtract(expected_fee).divide(price).format('0.00000000')
+              size = n(trade_balance).subtract(expected_fee).divide(price).format(s.product.asset_increment ? s.product.asset_increment : '0.00000000', Math.floor)
```

Comment:
Not sure how this is working.   at first glance it looks like 
if s.product.asset_increment is true it returns s.product.asset_increment else it returns '0.00000000'
are just checking if there is a value?  there always should be an increment.

Math.floor returns a function which is odd to me. will format execute the function using the value from the previous functional link?

Am I missing something here?  does n have its own format and its not utilizing the base string format or am I just ignorant of something.

Reply:
looking at this a little more your doing a test here because you introduced a new property to the products definition and can't expect to have all exchanges to have that property.  
The Math.floor question I still have though.

Reply:
Answered my question about the Math.
from the numbro docs: 
You can even round the result with your own rounding function (default is Math.round).

numbro(1000.234).formatCurrency("", Math.ceil);
//$1001

numbro(1000.234).formatCurrency("", Math.floor);
//$1000

Reply:
As above.